ksql-engine/src/test/java/io/confluent/ksql/integration/JsonFormatTest.java

```diff
@@ -224,6 +224,32 @@ public void testJsonStreamExtractor() throws Exception {
     ksqlEngine.terminateQuery(queryMetadata.getId(), true);
   }
 
+  @Test
+  public void testJsonStreamExtractorNested() throws Exception {
+
+    final String streamName = "JSONSTREAM";
+    final String queryString = String.format("CREATE STREAM %s AS SELECT EXTRACTJSONFIELD"
+                    + "(message, '$.log.logs[0].entry') "
+                    + "FROM %s;",
+            streamName, messageLogStream);
+
+    PersistentQueryMetadata queryMetadata =
+            (PersistentQueryMetadata) ksqlEngine.buildMultipleQueries(queryString, Collections.emptyMap()).get(0);
+    queryMetadata.getKafkaStreams().start();
+
+    Schema resultSchema = SchemaUtil
+            .removeImplicitRowTimeRowKeyFromSchema(metaStore.getSource(streamName).getSchema());
+
+    Map<String, GenericRow> expectedResults = new HashMap<>();
+    expectedResults.put("1", new GenericRow(Arrays.asList("first")));
+
+    Map<String, GenericRow> results = readNormalResults(streamName, resultSchema, expectedResults.size());
+
+    assertThat(results, equalTo(expectedResults));
+
+    ksqlEngine.terminateQuery(queryMetadata.getId(), true);
```

Comment:
This should probably be done in a `try{...}finally{...}' otherwise it won't run when the test fails